File: 프로그래머스/Level2/고득점 Kit/가장 큰 수-정렬.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

/*
테케 11번: numbers가 [0,0,0]인 경우 answer가 "000"임 => "0"으로 바꿔야함.
*/

bool comp(string a, string b){
    return a+b>b+a;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> v;
    
    for(int i=0; i<numbers.size(); i++){
        v.push_back(to_string(numbers[i]));
    }
    
    sort(v.begin(), v.end(), comp);
    
    for(int i=0; i<v.size(); i++){
        answer= answer + v[i];
    }
    
    if(answer[0]=='0'){
        answer = "0";
    }
    
    return answer;
}

File: 프로그래머스/Level2/고득점 Kit/가장 큰 수-정렬.py
def solution(numbers):
    answer = ''
    
    for i in range(0, len(numbers)) :
        numbers[i] = str(numbers[i])
    
    num_list = sorted(numbers, key=lambda x:x*3, reverse=True)
    
    for i in range(0, len(num_list)) :
        answer = answer + num_list[i]
    
    answer = str(int(answer))
    
    return answer